src/uniq.js

```diff
@@ -18,4 +20,39 @@ var uniqWith = require('./uniqWith');
  *      R.uniq([1, '1']);     //=> [1, '1']
  *      R.uniq([[42], [42]]); //=> [[42]]
  */
-module.exports = uniqWith(equals);
+module.exports = _curry1(function uniq(list) {
+  /* global Set */
+  var item, set, idx, len, items, uniqs;
+
+  /**
+   * Uses a native `Set` instance where possible for
+   * removing duplicate items. Items that implement
+   * the fantasy-land Setoid spec fallback to using
+   * `_contains` to support custom equality behaviour.
+   */
+  if (typeof Set === 'function') {
```

Comment:
You'll need an IIFE:

```javascript
module.exports = (function() {
  var uniq = function(...) { ... };
  return typeof Set === 'function' ? _curry1(uniq) : uniqWith(equals);
}());
```

See [__trim.js__][1] for an example of a module which uses this approach.


[1]: https://github.com/ramda/ramda/blob/v0.14.0/src/trim.js